README.md

```diff
@@ -8,98 +8,522 @@ Scheduling Abstraction Layer (SAL) is an abstraction layer initially developed a
+}
+```
+
+### 2.2. Deploying SAL as a Docker Container
+In this deployment approach, SAL runs inside a Docker container, providing a consistent environment across different systems. Management is handled via Docker commands or Docker Compose, with containerization isolating the application and its dependencies. While SAL can scale across multiple containers on the same machine, scalability is limited to a single-node setup unless additional tools are utilized. Docker manages networking, though more complex configurations may require manual setup. This method is ideal for consistent deployment across various environments, easier distribution, and meeting basic scalability needs.
+
+SAL can be deployed as a Docker container either by using a pre-built image or by building your own.
 
-### As Docker Container
 
-To use SAL as a Docker container, pull the public Docker image from DockerHub:
+#### 2.2.1. Using Pre-Built SAL Docker Images
+
+You can pull the latest or a specific version of the SAL Docker image from remote Docker repository [DockerHub](https://hub.docker.com/r/activeeon/sal/tags):
+- `activeeon/sal:dev`: The latest daily release of SAL.
+- `activeeon/sal:dev-YYYY-MM-DD`: A specific version of SAL released on a particular date. Replace YYYY-MM-DD with the desired date to pull that specific version.
+
+To pull an image:
 
 ```bash
-docker pull activeeon/sal
+docker pull activeeon/sal:dev
 ```
 
-## Usage
+#### 2.2.2. Creating a Custom SAL Docker Image:
 
-### Using SAL as a Microservice
+To create your own Docker image for SAL:
 
-To run SAL as a microservice, execute the following command:
+1. Clone the Docker repository:
 
 ```bash
-./gradlew bootRun
+git clone https://github.com/ow2-proactive/docker
+```
+
+2. Copy the built `.war` file:
+
+Copy the `.war` file generated in section 2.1.1  to the `docker/sal/artefacts` directory.
+
+3. Build the Docker image:
+
+Navigate to the `docker/sal` directory and build the image:
+```bash
+cd docker/sal
+docker build -t activeeon/sal:test -f ./Dockerfile --no-cache .
+```
+
+4. Publish the Docker image:
+
+```bash
+docker push activeeon/sal:test
+```
+
+#### 2.2.3. Run SAL as Docker Container:
+
+**Prerequisites:** Docker installed on your machine.
+
+1. Edit the Docker Compose File:
+
+* Open [docker-compose.yaml](https://github.com/ow2-proactive/scheduling-abstraction-layer/blob/master/docker/docker-compose.yaml)
+
+* Setup connection to the ProActive scheduler
+
+```bash
+sal:
+      #Set up connection to ProActive server (PWS)
+      PWS_URL: <CHANGE_ME>
+      PWS_USERNAME: <CHANGE_ME>
+      PWS_PASSWORD: <CHANGE_ME>
+```
+
+* Setup which SAL image will be used:
+
+```bash
+sal:
+#Set up image to be used for SAL from https://hub.docker.com/r/activeeon/sal/tags
+image: activeeon/sal:test
+```
+NOTE: It is possible to generate automatically image from `.war` file generated in section 2.1.1. In this case the image tag (e.g. test) should not exist in DockerHub repository.
```

Comment:
It is possible to generate automatically **an** image from **the** `.war` file generated in section 2.1.1